### ISSUE TITLE
Add AuthHubEvent For User Attribute Updated

### ISSUE THREAD

**Issue by LikemD** [aws-amplify/amplify-flutter#3749]
I find myself in need of a way to listen to changes to user attributes (custom attributes incl.) on my apps when changes are triggered by an external function. I realized that amplify-js supports this functionality but amplify-flutter doesn't, so i wanted to open this as a feature request.

**Comment by dgagnon** [comment]
As a workaround, we update the our data from the attributes on each signIn, but it does not pickup changes happening during a session and this has caused some issue in the past.
+1 for this feature.

**Comment by fjnoyp** [comment]
Hi @LikemD thanks for creating this issue.  To clarify, you'd like to listen to a new `userAttributesUpdated AuthHubEvent` so that you'd know when to call `Amplify.Auth.fetchUserAttributes()`?

**Comment by LikemD** [comment]
@fjnoyp Yes, that's right.

**Comment by LikemD** [comment]
@fjnoyp how long do you estimate that it'll take to release this feature? As this is a blocker for my product release

**Comment by Jordan-Nelson** [comment]
@LikemD - Can you clarify what you mean by "when changes are triggered by an external function"? Do you mean a function running outside of the current device (that is, on a sever or on another client)?

**Comment by LikemD** [comment]
@Jordan-Nelson Yes, that's right. Point is, i want to be able to listen for any change to an attribute in cognito

**Comment by mehrdad-shokri** [comment]
I would also appreciate if this event becomes possible to use
